drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -335,6 +381,72 @@ local function temp_attr_handler(driver, device, ib, response)
   end
 end
 
+local mired_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local color_temp_mireds = ib.data.value
+    if (color_temp_mireds < COLOR_TEMPERATURE_MIRED_MIN or color_temp_mireds > COLOR_TEMPERATURE_MIRED_MAX) then
+      device.log.warn_with({hub_logs = true}, string.format("Device reported a min or max color temperature %d mired outside of supported capability range", color_temp_mireds))
+      if (minOrMax == COLOR_TEMP_MIN) then
+        color_temp_mireds = COLOR_TEMPERATURE_MIRED_MIN
+      else
+        color_temp_mireds = COLOR_TEMPERATURE_MIRED_MAX
+      end
+    end
+    local temp_in_kelvin = mired_to_kelvin(color_temp_mireds)
+    set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..minOrMax, ib.endpoint_id, temp_in_kelvin)
+    local min = get_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, ib.endpoint_id)
+    local max = get_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, ib.endpoint_id)
+    if min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({ value = {minimum = min, maximum = max} }))
+        set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, ib.endpoint_id, nil)
+        set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, ib.endpoint_id, nil)
+      else
+        device.log.warn_with({hub_logs = true}, string.format("Device reported a min color temperature %d mired that is not lower than the reported max color temperature %d mired", min, max))
+      end
+    end
+  end
+end
+
+local level_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local current_level = ib.data.value
+    local lighting_endpoints = device:get_endpoints(clusters.LevelControl.ID, {feature_bitmap = clusters.LevelControl.FeatureMap.LIGHTING})
+    if ((#lighting_endpoints ~= 0 and current_level < SWITCH_LEVEL_LIGHTING_MIN) or current_level < SWITCH_LEVEL_MIN or current_level > SWITCH_LEVEL_MAX) then
```

Comment:
Can you add test cases for the different branches in this logic?

Also I think we need to be checking the endpoint reporting min or max attribute for the lighting endpoint. Right now I don't think it would work if a device happened to have multiple endpoints with the level control cluster but a mix of support for the lighting feature. That's probably unlikely but I think it should be pretty easy to do this check.